ding_unilogin/includes/ding_unilogin.admin.inc

```diff
@@ -6,71 +6,138 @@
     '#default_value' => variable_get('ding_unilogin_id', ''),
   );
 
   $form['login']['ding_unilogin_secret'] = array(
     '#type' => 'textfield',
-    '#title' => t('UNI•Login shared secret'),
+    '#title' => t('Unilogin shared secret'),
     '#description' => t('The shared secret used to calculate hashes.'),
     '#default_value' => variable_get('ding_unilogin_secret', ''),
   );
 
   $form['login']['ding_unilogin_global_logout'] = array(
     '#type' => 'checkbox',
     '#title' => t('Global logout'),
-    '#description' => t('Log out of UNI•Login when logging out of Drupal.'),
+    '#description' => t('Log out of Unilogin when logging out of Drupal.'),
     '#default_value' => variable_get('ding_unilogin_global_logout', 0),
   );
 
   // Settings for WS service general access.
   $form['ws'] = array(
     '#type' => 'fieldset',
-    '#title' => t('UNI•Login WS common'),
+    '#title' => t('Unilogin WS common'),
     '#description' => t('Common settings for the additional information web services.'),
   );
 
   $form['ws']['ding_unilogin_ws_user'] = array(
     '#type' => 'textfield',
-    '#title' => t('UNI•Login webservice user ID'),
+    '#title' => t('Unilogin webservice user ID'),
     '#default_value' => variable_get('ding_unilogin_ws_user', ''),
   );
 
   $form['ws']['ding_unilogin_ws_pass'] = array(
     '#type' => 'textfield',
-    '#title' => t('UNI•Login webservice password'),
+    '#title' => t('Unilogin webservice password'),
     '#default_value' => variable_get('ding_unilogin_ws_pass', ''),
   );
 
   // Settings for auth settings.
   $form['auth'] = array(
     '#type' => 'fieldset',
-    '#title' => t('UNI•Login Auth'),
+    '#title' => t('Unilogin Auth'),
     '#description' => t('Settings for the authentication web services.'),
   );
 
+  $configuration = _ding_unilogin_get_unilogin_helper()->getConfiguration();
+  $form['auth']['ding_unilogin_oidc'] = [
+    '#type' => 'fieldset',
+    '#title' => t('OIDC'),
+    '#tree' => TRUE,
+    '#description' => t('Settings for <a href="!url">Nyt Unilogin (OIDC)</a>', [
+      '!url' => 'https://viden.stil.dk/display/OFFSKOLELOGIN/OIDC',
+    ]),
+  ];
+
+  $form['auth']['ding_unilogin_oidc']['client_id'] = [
+    '#type' => 'textfield',
+    '#title' => t('Client ID'),
+    '#size' => 36,
+    '#default_value' => $configuration['client_id'] ?? NULL,
+    '#required' => TRUE,
+  ];
+
+  $form['auth']['ding_unilogin_oidc']['client_secret'] = [
+    '#type' => 'textfield',
+    '#title' => t('Client secret'),
+    '#size' => 36,
+    '#default_value' => $configuration['client_secret'] ?? NULL,
+    '#required' => TRUE,
+  ];
+
+  $form['auth']['ding_unilogin_oidc']['oidc_discovery_url'] = [
+    '#type' => 'textfield',
+    '#title' => t('OIDC discovery document URL'),
+    '#size' => 128,
+    '#default_value' => $configuration['oidc_discovery_url'] ?? NULL,
+    '#required' => TRUE,
+    '#description' => t('URL to your OIDC discovery document. Hints: !hints', [
```

Comment:
This would be the more "normal" Drupal 7 way to put links in descriptions in forms: 

`t('Look at Drupal documentation at <a href="@drupal-handbook">the Drupal Handbooks</a>.', array('@drupal-handbook' => 'http://drupal.org/handbooks'));`

Reply:
Like you do in line 129

Reply:
I know, but this is a clever trick to show a list of (two) useful URLs. In https://github.com/eReolen/ereolengo/blob/0b6c0d83576dc2608f97972cd7bad83e543b96b3/ding_unilogin/includes/ding_unilogin.admin.inc#L129 I always have one and exactly one URL.